app/services/payment_provider_customers/stripe_service.rb

```diff
@@ -72,6 +73,30 @@ def check_payment_method(payment_method_id)
       result.single_validation_failure!(field: :payment_method_id, error_code: 'value_is_invalid')
     end
 
+    def generate_checkout_url
+      return unless customer.organization.webhook_url?
+
+      res = Stripe::Checkout::Session.create(
+        checkout_link_params,
+        {
+          api_key:,
+        },
+      )
+
+      result.checkout_url = res['url']
+
+      SendWebhookJob.perform_later(
+        'customer.checkout_url_generated',
+        customer,
+        checkout_url: result.checkout_url,
+      )
+
+      result
+    rescue Stripe::InvalidRequestError, Stripe::PermissionError => e
+      deliver_error_webhook(e)
+      nil
```

Comment:
You should keep return `result` even without url, otherwise it will fail when calling `fail_with_error!` in the job.

Reply:
ok! Updated